openedx/frontend-app-authoring
# feat: simplify library home [FC-0062]

# Description

Simplifies the library UI/UX and code by using the same grid for the home, components and collections tab, also unifying the query used for collections and the react component used to display library content grids.

![image](https://github.com/user-attachments/assets/11ebee05-f53a-48e6-b34f-31fe7b7e20a0)


# Supporting Information

Implements: https://github.com/openedx/frontend-app-authoring/issues/1230 
Private Ref: [FAL-3918](https://tasks.opencraft.com/browse/FAL-3918)

# Testing Instructions

- Create a new Library
- Inside the new library create a new component and collection. 
    - Verify they both display in the "All Content" tab.
    - Verify that only the component appears in the "Components" tab
    - Verify that only the collection appears in the "Collections" tab
    - Verify that you can filter by the component type correctly
- Now add dozens of components by copy and pasting them
    - Verify that after a certain number if you refresh the page they are loaded in an infinite scroll
    - Verify that you can sort them and filter them correctly.
- Open the collection and verify you can both add new components and select components from the library by clicking "New" and "Existing Library Content"
![image](https://github.com/user-attachments/assets/62ebb8b1-f213-4ba1-b4db-e4a9723d0e98)
- Click on the search modal on top and then try to find XBlocks that are used in courses (we have to test this because the search results typing had to be changed a bit, but there should be no issue here).
![image](https://github.com/user-attachments/assets/6e66ad13-62fb-44fc-bee9-42ec47b452be)

- Now navigate to http://apps.local.openedx.io:2001/authoring/library/component-picker and verify components and collections render correctly.
